### PR TITLE
FT-69/ability to get a list of specializations

### PR DESCRIPTION
**Pull request by DenysLutskiy**

### CODE REVIEW

File: src/modules/specializations/specializations.resolver.ts [hunk 1]
```diff
@@ -42,4 +57,13 @@ export class SpecializationsResolver {
...
     return this.specializationsService.delete(specializationId);
   }
+
+  @ResolveField('doctors')
+  async getDoctor(@Parent() specialization: Specialization): Promise<Doctor[]> {
+    return (
+      await this.specializationsService.findOneById(specialization.id, {
+        relations: ['doctors'],
+      })
+    ).doctors;
```
Comment (d-smyrnov) [inline src/modules/specializations/specializations.resolver.ts]: Why don't you just query doctors by specialization id?

File: src/modules/specializations/specializations.service.ts [hunk 2]
```diff
@@ -66,4 +66,18 @@ export class SpecializationsService {
...
+
+  findAll(filter: undefined | string): Promise<Specialization[]> {
+    try {
+      if (filter) {
+        const text = ILike(`%${filter}%`);
+        return this.specializationRepository.find({
+          where: [{ name: text }],
+        });
+      }
+      return this.specializationRepository.find();
```
Comment (d-smyrnov) [inline src/modules/specializations/specializations.service.ts]: Can be simplified with a single return statement and dynamic condition object.

File: src/modules/specializations/specializations.resolver.ts (hunk 1 above)
Comment (DenysLutskiy) [reply src/modules/specializations/specializations.resolver.ts]: There is an extra task, to create FieldResolver for specialization.
Or what did you mean?

Comment (d-smyrnov) [reply src/modules/specializations/specializations.resolver.ts]: I meant that you can inject the doctor service and query doctors with specializationId filtration. No need to query specialization as you don't use it at all. 
(Not critical)

**Pull request merged**